### PR TITLE
[POCAM 62] update to drupal 9

### PR DESCRIPTION
**Pull request by lazysoundsystem**
https://humanitarian.atlassian.net/browse/POCAM-62

Updates composer and contrib modules (most of which already approved in https://github.com/UN-OCHA/cerf8/pull/950), and updates core to Drupal 9. 

(Custom modules had already been checked for Drupal9 compatibility before launch of Drupal8 version.)

Note: There are some changes to the default settings.php, which this PR ignores. Assuming we don't need to change anything about 'the UN-OCHA section' (`docker/settings.php`) to match those changes.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,29 +21,33 @@
     ],
     "require": {
         "php": ">=7.2",
```
Comment (cafuego) [inline composer.json:23]:
```suggestion
        "php": ">=7.4",
```

**Review by cafuego — APPROVED** [review]
Approved with the proviso to merge the PHP7.4 PR into this one first.

I've got a Drupal 9 settings file template in Ansible that is based off the one that ships with Drupal 9, not a copy of the one that we use for Drupal 8, so that *should* be happy :-)

**Comment by lazysoundsystem** [comment]
@cafuego If we're going to 7.4, should we change https://github.com/UN-OCHA/pocam8-site/blob/a5db819901094bd0f73de79572aa8669731d6f05/docker/Dockerfile#L1 and https://github.com/UN-OCHA/pocam8-site/blob/a5db819901094bd0f73de79572aa8669731d6f05/docker/Dockerfile#L11 too?

On https://hub.docker.com/r/unocha/php-k8s/tags?page=1&ordering=last_updated&name=7.4 I don't see a 7.4-stable.

**Comment by cafuego** [comment]
Totally, that's what https://github.com/UN-OCHA/pocam8-site/pull/79 is for :-) I was poking at the code in github via the browser and rather than spam more changes onto your PR I thought I'd branch off that and add them.

There is no -stable tag yet because nobody has run it for long enough (well, maybe Sesame) to warrant that.

**Comment by lazysoundsystem** [comment]
Aha, thanks @cafuego - that makes sense!

I noticed the Drupal version in Travis is still at 8.x and bumped it - there are some other version numbers there that could be bumped too: drush, coder, etc. https://github.com/UN-OCHA/pocam8-site/pull/78/commits/f0531c3ef2925d6e1d9ff51458a37a547940b135 (though I won't complain very loud while the tests still pass).

**Comment by cafuego** [comment]
I am 107% sure those version numbers in the travis env vars are there for hysterical rainsins and not at all used.

File: scripts/composer/ScriptHandler.php [hunk 2]
```diff
@@ -47,8 +47,8 @@ public static function createRequiredFiles(Event $event) {
       require_once $drupalRoot . '/core/includes/bootstrap.inc';
       require_once $drupalRoot . '/core/includes/install.inc';
       $settings['config_directories'] = [
-        CONFIG_SYNC_DIRECTORY => (object) [
-          'value' => Path::makeRelative($drupalFinder->getComposerRoot() . '/config/sync', $drupalRoot),
+        'config_sync_directory' => (object) [
```
Comment (lazysoundsystem) [inline scripts/composer/ScriptHandler.php:50]: Not sure what is needed here. This change makes the error go away, and I'm sure I saw it suggested somewhere, but note too https://github.com/UN-OCHA/iasc8/pull/504/files#diff-1006f32e007da05cefdc1061588471f49e9f2916d1a66b340b6b2510b4375898
and https://www.drupal.org/node/3018145

Comment (cafuego) [reply scripts/composer/ScriptHandler.php:50]: We use `root . '/config'` for all projects, so in *theory* that should work.

**Review by cafuego — APPROVED** [review]
YOLO!

**Comment by lazysoundsystem** [comment]
Closing in favour of https://github.com/UN-OCHA/pocam8-site/pull/83

**Pull request closed without merging**